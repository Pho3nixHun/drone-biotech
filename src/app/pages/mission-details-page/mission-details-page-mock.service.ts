import { Injectable } from '@angular/core';
import { Observable, BehaviorSubject } from 'rxjs';
import { MissionDetailsPageVM } from './mission-details-page.model';
import { MissionDetailsPageService } from './mission-details-page.service';

@Injectable({
    providedIn: 'root',
})
export class MissionDetailsPageMockService {
    public getVM(): Observable<MissionDetailsPageVM | null> {
        return this.vm$;
    }

    private readonly vm$: Observable<MissionDetailsPageVM | null> =
        vmSubject.asObservable();
}

const vmSubject = new BehaviorSubject<MissionDetailsPageVM | null>(null);

export const updateVMSubject = (vm: MissionDetailsPageVM | null) =>
    vmSubject.next(vm);

export const provideMissionDetailsPageMockService = () => ({
    provide: MissionDetailsPageService,
    useClass: MissionDetailsPageMockService,
});
