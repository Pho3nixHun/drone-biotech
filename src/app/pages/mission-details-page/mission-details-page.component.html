@if (vm(); as vm) {
    <app-page-layout>
        <app-google-map
            [vm]="vm.googleMapXVM"
            appDrawTargetAreas
            [targetAreas]="vm.googleMapXVM.targetAreaXVMs"
            appDrawEntryPoints
            [entryPoints]="vm.googleMapXVM.entryPointXVMs"
            class="h-96"
        />
    </app-page-layout>
}
