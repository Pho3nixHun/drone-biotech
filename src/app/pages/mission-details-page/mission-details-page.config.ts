import { ConfigProvider } from '@interfaces/config-provider';
import {
    POLYGON_OPTIONS,
    MAP_OPTIONS,
    ENTRY_POINT_MARKER_OPTIONS,
} from 'src/app/shared/providers/google-maps-provider';

export const missionDetailsPageConfig: ConfigProvider[] = [
    {
        provide: POLYGON_OPTIONS,
        useValue: {
            clickable: false,
            draggable: false,
            editable: false,
            fillOpacity: 0.5,
            strokeColor: 'blue',
            fillColor: 'blue',
        },
    },
    {
        provide: MAP_OPTIONS,
        useValue: {
            mapTypeId: 'roadmap',
            disableDoubleClickZoom: true,
            clickableIcons: false,
            mapId: 'DEMO_MAP_ID',
            zoom: 14,
            streetViewControl: false,
            mapTypeControlOptions: {
                position: 23.0,
            },
        },
    },
    {
        provide: ENTRY_POINT_MARKER_OPTIONS,
        useValue: {
            gmpDraggable: false,
        },
    },
];
