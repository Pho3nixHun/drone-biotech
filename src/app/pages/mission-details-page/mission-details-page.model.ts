import { Coordinates } from '@stores/location/location.model';

export interface MissionDetailsPageVM {
    mapFormControlXVM: MapFormControlXVM;
}

interface MapFormControlXVM {
    mapControl: MapControl;
}

export interface Mission {
    targetArea: Coordinates[];
    entryPoint: Coordinates;
}

export interface MapControl {
    targetArea: Coordinates[] | null;
    entryPoint: Coordinates | null;
}
