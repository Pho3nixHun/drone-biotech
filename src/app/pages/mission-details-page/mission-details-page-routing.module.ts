import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

export const routes: Routes = [
    {
        path: ':id',
        loadComponent: () =>
            import('./mission-details-page.component').then(
                (m) => m.MissionDetailsPageComponent
            ),
    },
];

@NgModule({
    imports: [RouterModule.forChild(routes)],
    exports: [RouterModule],
})
export class MissionDetailsPageRoutingModule {}
