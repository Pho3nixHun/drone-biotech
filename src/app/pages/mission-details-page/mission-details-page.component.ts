import { toSignal } from '@angular/core/rxjs-interop';
import { Component, inject } from '@angular/core';
import { MissionDetailsPageService } from './mission-details-page.service';
import { GoogleMapComponent } from '@components/google-map/google-map.component';
import { PageLayoutComponent } from '@components/page-layout/page-layout.component';
import { DrawEntryPointsDirective } from '@components/google-map/directives/draw-entry-points/draw-entry-points.directive';
import { DrawTargetAreasDirective } from '@components/google-map/directives/draw-target-areas/draw-target-areas.directive';

@Component({
    selector: 'app-mission-details-page',
    imports: [
        PageLayoutComponent,
        GoogleMapComponent,
        DrawTargetAreasDirective,
        DrawEntryPointsDirective,
    ],
    providers: [],
    templateUrl: './mission-details-page.component.html',
})
export class MissionDetailsPageComponent {
    private readonly missionDetailsPageService = inject(
        MissionDetailsPageService
    );

    protected readonly vm = toSignal(this.missionDetailsPageService.getVM());
}
