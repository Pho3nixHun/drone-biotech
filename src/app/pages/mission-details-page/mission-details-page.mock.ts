import { MissionDetailsPageVM } from './mission-details-page.model';

export const missionDetailsPageVM: MissionDetailsPageVM = {
    googleMapXVM: {
        center: { lat: 47.277, lng: 21.2343 },
        mapOptions: {
            mapTypeId: 'roadmap',
            disableDoubleClickZoom: true,
            clickableIcons: false,
            mapId: 'DEMO_MAP_ID',
            zoom: 14,
            streetViewControl: false,
        },
        entryPointXVMs: [
            {
                options: {},
                position: { lat: 47.277, lng: 21.2343 },
            },
        ],
        targetAreaXVMs: [
            {
                options: {
                    clickable: false,
                    draggable: false,
                    editable: false,
                    fillOpacity: 0.5,
                    strokeColor: 'green',
                    fillColor: 'green',
                },
                coordinates: [
                    { lat: 47.277, lng: 21.2345 },
                    { lat: 47.276, lng: 21.2345 },
                    { lat: 47.276, lng: 21.233 },
                    { lat: 47.277, lng: 21.233 },
                ],
            },
        ],
    },
};
