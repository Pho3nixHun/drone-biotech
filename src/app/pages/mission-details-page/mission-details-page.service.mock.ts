import { Injectable, Signal, signal } from '@angular/core';
import { MissionDetailsPageService } from './mission-details-page.service';
import {
    MessageItemXVM,
    MissionDetailsPageVM,
} from './mission-details-page.model';
import { missionDetailsPageDefaultVM } from './mission-details-page.mock';
import { noop } from 'rxjs';
import { ProgressLogItemVM } from './components/progress-update-form-control/components/progress-frame/components/progress-log-item-list/components/progress-log-item/progress-log-item.model';

@Injectable({
    providedIn: 'root',
})
export class MissionDetailsPageServiceMock {
    public getVM(): Signal<MissionDetailsPageVM> {
        return vmSignal;
    }

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    public sendMessage = (item: MessageItemXVM) => noop;

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    public addLogItem = (item: ProgressLogItemVM) => noop;
}

export const provideMockMissionDetailsPageService = () => ({
    provide: MissionDetailsPageService,
    useClass: MissionDetailsPageServiceMock,
});

const vmSignal = signal<MissionDetailsPageVM>(missionDetailsPageDefaultVM);

export const updateVMSignal = (vm: MissionDetailsPageVM) => vmSignal.set(vm);
