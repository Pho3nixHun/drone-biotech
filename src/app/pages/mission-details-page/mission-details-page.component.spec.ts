import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MissionDetailsPageComponent } from './mission-details-page.component';
import { missionDetailsPageVM } from './mission-details-page.mock';
import {
    provideMissionDetailsPageMockService,
    updateVMSubject,
} from './mission-details-page-mock.service';

describe('MissionDetailsPageComponent', () => {
    let compiled: HTMLElement;
    let fixture: ComponentFixture<MissionDetailsPageComponent>;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [MissionDetailsPageComponent],
            providers: [provideMissionDetailsPageMockService()],
        }).compileComponents();

        fixture = TestBed.createComponent(MissionDetailsPageComponent);
        compiled = fixture.debugElement.nativeElement;
    });
    // Snapshot testing
    it('should render the template correctly', () => {
        // Arrange
        updateVMSubject(missionDetailsPageVM);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
