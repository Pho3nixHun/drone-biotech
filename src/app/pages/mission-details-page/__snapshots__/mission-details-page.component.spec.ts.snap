// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionDetailsPageComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-page-layout>
    <section
      class="flex flex-col gap-y-10 mx-5 sm:mx-20 lg:mx-32"
    >
      <app-google-map
        appdrawentrypoints=""
        appdrawtargetareas=""
        class="h-96"
        ng-reflect-entry-points="[object Object]"
        ng-reflect-target-areas="[object Object]"
        ng-reflect-vm="[object Object]"
      >
        <div
          appmapinitializer=""
          ng-reflect-map-center="[object Object]"
          ng-reflect-map-options="[object Object]"
          ng-reflect-map-signal="[Signal: null]"
          style="height: 100%;"
        />
      </app-google-map>
    </section>
  </app-page-layout>
  <!--container-->
</div>
`;
